### DISCUSSION TITLE
Understanding no-floating-promises

### DISCUSSION THREAD

**Discussion by GZLiew** [typescript-eslint/typescript-eslint#8046] [Technical Discussions]
### Body

Hi I added the `@typescript-eslint/no-floating-promises` rule to my code base but why does the following code not error ?

```json
    '@typescript-eslint/no-floating-promises': 'error',
```

```ts
export const sleep = (ms = 3000) =>
  new Promise((resolve) => setTimeout(resolve, ms));

const a =  sleep();
```

but calling only sleep() errors

```ts
    sleep(); // Promises must be awaited
```

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
The rule assumes your variable will be used and handled appropriately.
Unused variables can be caught using the `no-unused-vars` rule.

**Comment by JoshuaKGoldberg** [comment]
Looks like this was a straightforward resolution 🙂. Thanks all!

Filed https://github.com/typescript-eslint/typescript-eslint/issues/8088 to improve docs around this.
